app/views/spree/admin/variants/_autocomplete.js.erb

```diff
@@ -29,3 +29,43 @@
     </div>
   </div>
 </script>
+<script type='text/template' id='variant_autocomplete_stock_template'>
+  <fieldset>
+    <legend align="center"><%= Spree.t(:select_stock) %></legend>
+      <table class="stock-levels" data-hook="stock-levels">
+        <colgroup>
+          <col style="width: 30%;" />
+          <col style="width: 40%;" />
+          <col style="width: 20%;" />
+          <col style="width: 10%;" />
+        </colgroup>
+        <thead>
+          <th><%= Spree.t(:location) %></th>
+          <th><%= Spree.t(:count_on_hand) %></th>
+          <th><%= Spree.t(:quantity) %></th>
+          <th class="actions"></th>
+        </thead>
+        <tbody>
+          {{#each variant.stock_items}}
+            <tr>
+              <td>{{this.stock_location_name}}</td>
+              {{#if this.available}}
+                <td>
+                  {{this.count_on_hand}}
+                  {{#if this.backorderable}} (backorders allowed) {{/if}}
+                </td>
+                <td>
+                  <input class="quantity" id="stock_item_quantity" data-stock-location-id="{{this.stock_location_id}}" type="number" min="1" value="1">
+                </td>
+                <td class="actions">
+                  <button class="add_variant no-text icon-plus icon_link with-tip" data-stock-location-id="{{this.stock_location_id}}" title="Add" data-action="add"></button>
+                </td>
+              {{else}}
+                <td><%= Spree.t(:out_of_stock) %></td>
```

Comment:
all Spree.t entries should be replaced by normal lazy loaded entries (may need to add these to our en.yml file).